src/providers/twitch/TwitchChannel.cpp

```diff
@@ -676,26 +681,40 @@ void TwitchChannel::refreshChatters()
         }
     }
 
-    // get viewer list
-    NetworkRequest("https://tmi.twitch.tv/group/user/" + this->getName() +
-                   "/chatters")
-
-        .onSuccess(
-            [this, weak = weakOf<Channel>(this)](auto result) -> Outcome {
-                // channel still exists?
-                auto shared = weak.lock();
-                if (!shared)
-                    return Failure;
+    if (getSettings()->lowercaseUsernames)
+    {
+        // get viewer list
+        NetworkRequest("https://tmi.twitch.tv/group/user/" + this->getName() +
+                       "/chatters")
+
+            .onSuccess(
+                [this, weak = weakOf<Channel>(this)](auto result) -> Outcome {
+                    // channel still exists?
+                    auto shared = weak.lock();
+                    if (!shared)
+                        return Failure;
 
-                auto pair = parseChatters(result.parseJson());
-                if (pair.first)
-                {
-                    this->setChatters(std::move(pair.second));
-                }
+                    auto pair = parseChatters(result.parseJson());
+                    if (pair.first)
+                    {
+                        this->setChatters(std::move(pair.second));
+                    }
 
-                return pair.first;
-            })
-        .execute();
+                    return pair.first;
+                })
+            .execute();
+    }
+    else
+    {
+        getHelix()->getUserByName(
+            this->getName(),
+            [this](const auto &user) {
+                this->addRecentChatter(user.displayName);
+            },
+            [] {
+                // on failure
+            });
+    }
```

Comment:
This piece of code leads to quite different behavior depending on whether the setting is enabled or not.

If it is enabled, the behavior is the same as currently, i.e. users connected to the channel are fetched directly from TMI. This includes users that have not sent a message since you connected to the channel.

If the setting is not enabled, the piece of code will only add the channel owner's display name to the chatters list.
The problem¹  with this approach is that you can only autocomplete chatters that already typed something since you joined the chat room (and users from `recent-messages`). Users that connected to TMI but have not sent a message since you joined the channel are not considered. This feels inconsistent.

----
¹: It is arguable if people *actually* consider this a problem. In my opinion, the behavior should feels consistent regardless of whether the settings is applied or not.